### ISSUE TITLE
Matching for \Z does not work

### ISSUE THREAD

**Issue by hoanbui29** [fancy-regex/fancy-regex#147]
My regex: Regex::new(r"(?<=do\(\)|\A)[\s\S]*?(?=don\'t\(\)|\Z)").unwrap();

Error: thread 'main' panicked at src/main.rs:21:81:
called `Result::unwrap()` on an `Err` value: ParseError(36, InvalidEscape("\\Z"))

It seems like matching for end of string (\Z) does not work as expected.

![CleanShot 2024-12-04 at 10 58 22@2x](https://github.com/user-attachments/assets/3a0d1f60-9807-4ec4-a750-24171dd7b492)

**Comment by keith-hall** [comment]
Indeed, [Oniguruma supports it](https://github.com/kkos/oniguruma/blob/c4a55f8363dfd4c60320000871c6de44b28615fc/doc/RE#L180) but the `regex` crate and `fancy-regex`  only support `\z`. 
Here is how `fancy-regex` currently handles `\z`, in case it helps someone work out where to start with adding support for `\Z`:
https://github.com/fancy-regex/fancy-regex/blob/main/src/parse.rs#L352-L353
https://github.com/fancy-regex/fancy-regex/blob/main/src/vm.rs#L495
